lib/src/scss-shared/_nx-form-validation.scss

```diff
@@ -5,20 +5,70 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 @import 'nx-variables';
+@import 'nx-container-helpers';
 
 @mixin valid() {
-  border-left-color: $nx-valid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
```

Comment:
I'll continue to crusade against `:not`.  I suspect there's a way to avoid using it for this.

Reply:
> I'll continue to crusade against `:not`. I suspect there's a way to avoid using it for this.

While historically speaking crusades have had mixed results in this case we have achieved victory: https://github.com/sonatype/sonatype-react-shared-components/pull/57/commits/600837d63c526088a976a7de4384b2426cb07531